### PR TITLE
Adding null check for API Product Username and Password

### PR DESCRIPTION
**Pull request by sanjulamadurapperuma**
Adding null check for API Product Username and Password. Will potentially fix API Product Creation Test failures.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/template/SecurityConfigContext.java [hunk 1]
```diff
@@ -173,10 +173,13 @@ public VelocityContext getContext() {
                         endpointSecurityModel.setType(endpointSecurityEntry.getValue().getType());
                         endpointSecurityModel
                                 .setAdditionalProperties(endpointSecurityEntry.getValue().getAdditionalProperties());
-                        endpointSecurityModel.setBase64EncodedPassword(new String(Base64.encodeBase64(
-                                endpointSecurityModel.getUsername().concat(":")
-                                        .concat(endpointSecurityModel.getPassword())
-                                        .getBytes())));
+                        if (endpointSecurityModel.getUsername() != null &&
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/template/SecurityConfigContext.java:176]: StringUtils.

Comment (sanjulamadurapperuma) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/template/SecurityConfigContext.java]: Changed it to StringUtils

**Comment by Arshardh** [comment]
Please get this merged soon

**Comment by sanjulamadurapperuma** [comment]
Fixed in [wso2/carbon-apimgt#8874](https://github.com/wso2/carbon-apimgt/pull/8874/files#diff-52a850fdd5549b06f7f7609bce53cf9bR176). Hence, closing this PR.

**Pull request closed without merging**